packages/react-server-module-tagger/index.js

```diff
@@ -0,0 +1,129 @@
+var isWindows = ('win32' === process.platform);
+
+/**
+ * A util function for tagging modules.  Expects to find data on its prototype
+ * as follows
+ *     {
+ *         file: {
+ *             path: the path to the file to tag
+ *         }
+ *         config: {
+ *             trim: the string to trim off the front of the logger name
+ *         }
+ *     }
+ * @example
+ *     const file = '/path/to/my/file.js';
+ *     loggerSpec.bind({file, { trim: 'path.to.' }})(file, '({label: "foo"})')
+ *     // returns '{\"label\":\"foo\",\"name\":\"my.file\",\"color\":{\"server\":87,\"client\":\"rgb(42,212,212)\"}}'
+ * @param  {String} fullMatch May also be provided as this.file.path, the path to the file
+ * @param  {String} optString The label to add to the module tag, in the form '({label:"$label"})'
+ * @return {String}           A json object containing a module identifier
+ */
+module.exports = function(fullMatch, optString){
+	var fn   = this.file && this.file.path ? this.file.path : fullMatch
+	,   trim = this.config.trim || ''
+	,   basePath = this.config.basePath || ''
+	,   opts = {}
+
+	if (fn.indexOf(basePath) !== 0) {
+		throw new Error("Unable to handle " + basePath + " for " + fn);
+	}
+
+	if (optString) {
+		// The slash replacement here is so we don't choke on example
+		// loggers in comments.
+		opts = new Function("return "+optString.replace(/^\/\//mg,''))(); // eslint-disable-line no-new-func
+	}
+
+	opts.name  = getName  (fn, opts, trim, basePath);
+	opts.color = getColor (opts);
+
+	return JSON.stringify(opts);
+}
```

Comment:
Really, a nicer interface would be to accept the context object as an argument and return `opts` as an object.  The binding and serialization seems more relevant to the calling environment in the gulp module tagger.